### PR TITLE
Sean/add annotations

### PR DESCRIPTION
**Pull request by Sean-Holcomb**
Tested by filtering on annotations using requests like this on bolt-1:

/model/aggregatedCostModel?window=1d&offset=1m&aggregation=namespace&allocateIdle=true&timeSeries=true&annotations=test_kubecost_annotation_instance%3Dkubecost&efficiency=true&aggregationSubfield=&sharedNamespaces=&sharedSplit=weighted&sharedLabelNames=&sharedLabelValues=&sharedSplit=weighted&req=635724

Additional annotations to filter on can be found in the bolt-1 prometheus dashboard by querying kube_pod_annotations

### CODE REVIEW

**Comment by dwbrown2** [comment]
Nice! Does this endpoint now support `aggregation=annotation`? 

Can we do a quick doc on how to enable these + metrics plus how to use them?

**Comment by Sean-Holcomb** [comment]
Thanks @mbolt35 for multi-filter bug fix

**Comment by Sean-Holcomb** [comment]
> Nice! Does this endpoint now support `aggregation=annotation`?
> 
> Can we do a quick doc on how to enable these + metrics plus how to use them?
Pr for aggregation on annotations
https://github.com/kubecost/cost-model/pull/657

Working with Ajay on adding in documentation

**Review by nikovacevic — CHANGES_REQUESTED** [review]
Good work on this so far. I haven't manually tested yet, but if we address the few small things I've noted and get it thoroughly tested, then I think this will be ready to approve and merge. Let me know if you have any questions.

File: pkg/costmodel/aggregation.go [hunk 1]
```diff
@@ -1842,10 +1912,8 @@ func (a *Accesses) AggregateCostModelHandler(w http.ResponseWriter, r *http.Requ
 	namespace := r.URL.Query().Get("namespace")
 	cluster := r.URL.Query().Get("cluster")
 	labels := r.URL.Query().Get("labels")
+	annotations := r.URL.Query().Get("annotations")
 	podprefix := r.URL.Query().Get("podprefix")
-	labelArray := strings.Split(labels, "=")
-	labelArray[0] = strings.ReplaceAll(labelArray[0], "-", "_")
-	labels = strings.Join(labelArray, "=")
```
[the remark this answers is not in the document]
Comment (Sean-Holcomb) [reply pkg/costmodel/aggregation.go:1848]: label sanitization happens later in the process and these particular snippet doesn't account for multiple label filters that are comma separated like the later one does

**Comment by AjayTripathy** [comment]
This LGTM, but I'll let @nikovacevic give the final review.

File: pkg/costmodel/aggregation.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbolt35) [reply pkg/costmodel/aggregation.go:1864]: @nikovacevic Yeah, this code is completely unnecessary.

File: pkg/costmodel/costmodel.go [hunk 2]
```diff
@@ -85,6 +85,7 @@ type CostData struct {
 	GPUReq          []*util.Vector               `json:"gpureq,omitempty"`
 	PVCData         []*PersistentVolumeClaimData `json:"pvcData,omitempty"`
 	NetworkData     []*util.Vector               `json:"network,omitempty"`
+	Annotations     map[string]string            `json:"Annotations,omitempty"`
```
Comment (mbolt35) [inline pkg/costmodel/costmodel.go:88]: `json:"annotations,omitempty"` camel case for json properties

**Review by mbolt35 — APPROVED** [review]
LGTM after the one small change to the annotations json property.

File: pkg/costmodel/costmodel.go (hunk 2 above)
Comment (Sean-Holcomb) [reply pkg/costmodel/costmodel.go]: updated

File: pkg/costmodel/costmodel.go [hunk 3]
```diff
@@ -1687,6 +1729,19 @@ func (cm *CostModel) costDataRange(cli prometheusClient.Client, cp costAnalyzerC
 		klog.V(1).Infof("Unable to get Pod Labels for Metrics: %s", err.Error())
 	}
 
+	nsAnnotations, err := GetNamespaceAnnotationsMetrics(resNSAnnotations, clusterID)
+	if err != nil {
+		klog.V(1).Infof("Unable to get Namespace Annotations for Metrics: %s", err.Error())
+	}
+	if nsAnnotations != nil {
+		appendNamespaceLabels(namespaceAnnotationsMapping, nsAnnotations)
```
Comment (mbolt35) [inline pkg/costmodel/costmodel.go:1737]: We can probably rename `appendNamespaceLabels` to just `mergeStringMap` or something :)

**Comment by Sean-Holcomb** [comment]
This PR is a subset of this other pr

kubecost/cost-model#657

**Pull request closed without merging**